### PR TITLE
feat: add ignoreFields option to regenerate method

### PR DESCRIPTION
**Pull request by RafaelGSS**
You can pass an array of fields that should be kept when the session is regenerated

### CODE REVIEW

File: types/types.d.ts [hunk 1]
```diff
@@ -38,9 +38,13 @@ interface SessionData extends ExpressSessionData {
 
   /**
    * Regenerates the session by generating a new `sessionId`.
+   *
+   * ignoreFields specifies which fields should be kept in the new session object.
    */
   regenerate(callback: Callback): void;
+  regenerate(callback: Callback, ignoreFields: string[]): void;
```
Comment (Eomm) [inline types/types.d.ts:45]: Isn't the callback as the first parameter odd?

File: README.md [hunk 2]
```diff
@@ -118,7 +118,7 @@ Allows to destroy the session in the store. If you do not pass a callback, a Pro
 
 Updates the `expires` property of the session's cookie.
 
-#### Session#regenerate(callback)
+#### Session#regenerate(callback, [ignoreFields])
```
Comment (climba03003) [inline README.md:121]: `callback` should place at the last field.
We can check if the first field is `Array` or `Function` to prevent breaking changes.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: types/types.d.ts (hunk 1 above)
Comment (RafaelGSS) [reply types/types.d.ts:45]: yes, but it will not introduce a breaking chance

**Review by climba03003 — APPROVED** [review]

File: lib/session.js [hunk 3]
```diff
@@ -85,6 +96,12 @@ module.exports = class Session {
           this[cookieSignerKey]
         )
 
+        if (Array.isArray(keys)) {
+          for (const key of keys) {
+            session.set(key, this[key])
+          }
+        }
```
Comment (mcollina) [inline lib/session.js:103]: I don't really like this code duplication. I think we should move the whole `new Session` block outside of the if-else.

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm but CI is failing

**Comment by RafaelGSS** [comment]
> lgtm but CI is failing

I've run it several times locally and it passes. I've re-run it on actions and it's passing as well. It doesn't seem like a test flaky, but an environment flaky (gh actions).

**Pull request merged**